web/ui/shutdownvis/src/app.js

```diff
@@ -0,0 +1,393 @@
+import {get, debounce} from "./utils";
+import {buildControls} from "./controls";
+import {buildLegend} from "./legend";
+import {buildShutdownList} from "./shutdownlist";
+import {buildGraph} from "./graph";
+
+// colors come in pairs, one for actual data, 
+// the other for calculated data
+const COLORS = [
+    ["#a6cee3", "#1f78b4"],
+    ["#b2df8a", "#33a02c"],
+    ["#cab2d6", "#6a3d9a"]
+];
+
+// try to recalculate prediction values
+// as the slider is adjusted.
+const CALCULATE_ON_INPUT = false;
+const PREDICTION_DEBOUNCE = 200;
+
+const TARGET_DP_COUNT = 10000;
+
+let state = {
+    model: {
+        margin: 15e10,
+        lookAhead: 20 * 60 * 1000,
+        theWindow: 30 * 60 * 1000 
+    },
+    tsdb: {
+        url: "http://uiboss:4242",
+        tenantIDs: ["8noa8p8e84m3stayz1zqr3egx"],
+        start: new Date(new Date().getTime() - 1000 * 60 * 60 * 24 * 7),
```

Comment:
The URL and tenant id is hardcoded for now because the user will always have to replace it with their own values. In the future we could try to get those values from serviced.